sdk/storage/azure-storage-common/src/main/java/com/azure/storage/common/StorageSharedKeyCredential.java

```diff
@@ -156,7 +157,8 @@ private String getAdditionalXmsHeaders(Map<String, String> headers) {
             return "";
         }
 
-        Collections.sort(xmsHeaderNameArray);
+        /* Culture-sensitive word sort */
+        Collections.sort(xmsHeaderNameArray, Collator.getInstance());
```

Comment:
Should we use Collator.getInstance(Locale)  ? Otherwise this will use local machine or JVM default which might not be what we expect. I guess en_US would be safe choice.

Reply:
good point, yeah I wasn't too sure which one to pin it to if I did. There's also Locale.ROOT, Locale.US and Locale.ENGLISH

Reply:
Javadoc of Locale.ROOT looks promising - I'd give that a try. If that doesn't work Locale.US.